src/coreclr/src/inc/corinfo.h

```diff
@@ -1012,6 +1012,8 @@ enum CorInfoClassId
     CLASSID_STRING,
     CLASSID_ARGUMENT_HANDLE,
     CLASSID_RUNTIME_TYPE,
+    CLASSID_ENUM,
+    CLASSID___CANON,
```

Comment:
`Canon` is an implementation detail of the EE side of the type system. It would be really nice to avoid it leaking it to the JIT. Different implementation of the type system can have multiple Canons, or no Cannons at all.

Reply:
I agree, in mono we have T_REF and gsharedvt